bones/passwordBone.py

```diff
@@ -60,23 +61,41 @@ class passwordBone( stringBone ):
 	def isInvalid(self, value):
 		if not value:
 			return False
-		if len(value)<self.minPasswordLength:
-			return "Password to short"
-		# Run our passwort test suite
+
+		if len(value) < self.minPasswordLength:
+			return _("The entered password is to short - it requires at least {{length}} characters.",
+			            length=self.minPasswordLength)
+
+		# Run our password test suite
 		testResults = []
 		for test in self.passwordTests:
 			testResults.append(test(value))
-		if sum(testResults)<self.passwordTestThreshold:
-			return("Your password isn't strong enough!")
+
+		if sum(testResults) < self.passwordTestThreshold:
+			return _("The entered password is too weak.")
+
 		return False
 
+	def fromClient( self, valuesCache, name, data ):
```

Comment:
I think we can simplify that. Check for "not value" first (and return "No value entered" directly), and then test for isInvalid (return err) else store value. 
With the current code, err is always False if no value is set; so the expression "not value and not err" can be shortened

Reply:
You're right, I've fixed it now and hope this fits your recommendations.